spring-cloud-deployer-local/src/test/java/org/springframework/cloud/deployer/spi/local/LocalTaskLauncherIntegrationTests.java

```diff
@@ -80,6 +94,31 @@ protected String randomName() {
 		}
 	}
 
+	@Test
+	public void testPassingServerPortViaCommandLineArgs() throws InterruptedException {
+		Map<String, String> appProperties = new HashMap();
+		appProperties.put("killDelay", "0");
+		appProperties.put("exitCode", "0");
+
+		AppDefinition definition = new AppDefinition(this.randomName(), appProperties);
+
+		Resource resource = this.testApplication();
+
+		List<String> commandLineArgs = new ArrayList<>(1);
+		commandLineArgs.add(LocalTaskLauncher.SERVER_PORT_KEY_PREFIX + SocketUtils.findAvailableTcpPort(LocalTaskLauncher.DEFAULT_SERVER_PORT));
```

Comment:
This test doesn't really validate that the app got the `server.port`....  Commenting the line out makes this test equivalent to `testSimpleLaunch` in AbstractTaskLauncherIntegrationTests.  Prob would need to hit the actuator endpoint to make sure the `server.port` value is what was expectedat an integration test level.

Reply:
There probably isn't Actuator in the app which is why I didn't go further (if I'm wrong, then I agree, but we usually don't include it in tasks).  The test does fail in the way the build was failing when the old code is in place (the new code fixes the test).

Reply:
I'll add a unit test for calcServerPort to ensure the behavior/overides and keep this one.  Moving it up to AbstractTaskLauncherIntegration tests may end up breaking it on other platforms which we don't need to deal with now.  I'll create an issue to investigate moving it to the base class.

Reply:
Is this base class used via the other deployers?  I was running under the assumption that code common across deployer impls would live in the SPI project...